application/client/src/app/ui/views/toolbar/search/input/component.ts

```diff
@@ -83,16 +89,30 @@ export class ViewSearchInput
 
     public ngAfterViewInit(): void {
         this.input.bind(this.searchInputRef.nativeElement, this.recentPanelRef);
-        this.input.actions.accept.subscribe(() => {
-            const filter = this.input.asFilter();
-            this.recent.update(filter.filter);
+        this.input.actions.accept.subscribe((flagUpdate: boolean) => {
+            let filter: IFilter;
+            if (flagUpdate) {
+                if (this.active === undefined) {
+                    return;
+                }
+                filter = {
+                    filter: this.active.filter.filter,
+                    flags: this.input.flags,
+                };
+            } else {
+                filter = this.input.asFilter();
+                this.recent.update(filter.filter);
+            }
```

Comment:
No need to add extra-flag to `accept` event. If you are changing a flags with active search, you will have `this.active !== undefined` it will give you understanding - no needs to save it as recent. Well
```
this.acrive === undefined && this.recent.update(filter.filter);
```
should work instead of additional flags, but check me please for sure.